crates/noirc_frontend/src/parser/mod.rs

```diff
@@ -164,14 +164,10 @@ where
 {
     chumsky::prelude::none_of(targets)
         .repeated()
-        .ignore_then(one_of(too_far.clone()).rewind())
-        .try_map(move |peek, span| {
-            if too_far.get_iter().any(|t| t == peek) {
-                // This error will never be shown to the user
-                Err(ParserError::with_reason(String::new(), span))
-            } else {
-                Ok(Recoverable::error(span))
-            }
+        .ignore_then(one_of(too_far).or_not().rewind())
+        .try_map(move |output, span| match output {
+            Some(_) => Err(ParserError::with_reason(String::new(), span)),
```

Comment:
Lets keep the "This error will never be shown to the user" comment to explain why we're returning an empty string for the error here

Reply:
accidentally left that out, will add it back, thanks for pointing out